test/dsets.c

```diff
@@ -12514,10 +12514,19 @@ test_bt2_hdr_fd(const char *env_h5_driver, hid_t fapl)
     hid_t             dcpl = -1;
     hid_t             msid = -1;
     H5D_chunk_index_t idx_type;
-    const hsize_t     shape[2]    = {8, 8};
-    const hsize_t     maxshape[2] = {H5S_UNLIMITED, H5S_UNLIMITED};
-    const hsize_t     chunk[2]    = {8, 8};
-    const int         buffer[8]   = {0, 1, 2, 3, 4, 5, 6, 7};
+    const hsize_t     shape[2]     = {8, 8};
+    const hsize_t     maxshape[2]  = {H5S_UNLIMITED, H5S_UNLIMITED};
+    const hsize_t     chunk[2]     = {8, 8};
+    const int         buffer[8][8] = {
```

Comment:
Test was previously writing to an 8x8 dataset with an 8x1 array, underflowing into the following variables and causing a stack overflow.